cli/tests/pcluster/schemas/test_cluster_schema.py

```
@@ -212,6 +214,107 @@ def test_head_node_root_volume_schema(mocker, config_dict, failure_message):
 }
 
 
+@pytest.mark.parametrize(
+    "config_dict, failure_message",
+    [
+        # Failures
+        ({"OnNodeUpdating": "test"}, "Unknown field"),
+        ({"OnNodeStart": "test", "OnNodeConfigured": "test", "OnNodeUpdated": "test"}, "Invalid input type."),
+        # Successes
+        (
+            {
+                "OnNodeStart": {"Script": "test", "Args": ["1", "2"]},
+                "OnNodeConfigured": {"Script": "test2", "Args": ["3", "4"]},
+                "OnNodeUpdated": {"Script": "test3", "Args": ["5", "6"]},
```

minor: let's add a failing test with an additional field or a wrong name "e.g. ScriptWrong in place of String, that should fail with Unknown field.

---

Great catch!